ui/sd_internal/task_manager.py

```diff
@@ -268,6 +273,11 @@ def thread_render(device):
             return
         task = thread_get_next_task()
         if task is None:
+            if runtime.thread_data.device == 'cpu' and is_alive() > 1 and hasattr(runtime.thread_data, 'lastActive') and time.time() - runtime.thread_data.lastActive > CPU_UNLOAD_TIMEOUT:
+                # GPUs present and CPU is idle. Unload resources.
+                runtime.unload_models()
```

Comment:
What happens if a user requests 'CPU' again in a subsequent request?

Reply:
Noting as long as it's within 4 minutes.
lastActive update each time a new task is sent to it.
So models unloading should not trigger unless CPU is alone and idle for more than CPU_UNLOAD_TIMEOUT

Reply:
What I mean is, if the user makes a request after the model has unloaded (takes a lunch break and comes back)? Won't it just crash?